plugins/woocommerce-admin/client/marketplace/components/product-list-content/product-list-content.tsx

```diff
@@ -45,6 +49,12 @@ export default function ProductListContent( props: {
 							  ] )
 							: '',
 						price: product.price,
+						...( product.label && { label: product.label } ),
+						...( props.group && { group: props.group } ),
+						...( props.searchTerm && {
```

Comment:
Why are we using this pattern?
Wouldn't it be enough to do `searchTerm: props.searchTerm`?

Reply:
I'll answer, since this is my part of the code. That still leaves a blank prop value in the final object. This pattern ensures that only props with values are in the final object. It's not a huge concern, but an earlier comment highlighted that it might be better to only keep props with values.

Reply:
Ignore me, for some reason I thought that the key would be excluded.

Still if we [would use a helper function](https://github.com/woocommerce/woocommerce/pull/41365#discussion_r1395238150) we could do it there and avoid having so much code.

Reply:
Yeah, definitely like that idea. Good to note for future and at least Cem can rest knowing it was me who coded this spaghetti 😂